tools/ab_test

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+# -*- shell-script -*-
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -eu -o pipefail
+shopt -s lastpipe
+
+source tools/functions
+
+if [ $# -lt 3 ]; then
+    cat <<EOF
+$0 <a revision> <b revision> <A/B test glob> [pytest args]
+
+    Example: $0 f6c28a6e2c50eaf021130c3f4e59cbd8daf0ea12 bb2fa406e12e7c49f48fce770b57fe6ec856d263 integration_tests/performance/test_snapshot_ab.py
+
+    Runs the specified A/B test with the given revisions. This is useful for running ad-hoc A/B-test to investigate
+    performance impact of a specific commit range by running this script as a buildkite step.
+EOF
+    exit 1
+fi
+
+function get_binaries {
+  revision=$1
+
+  say "Getting binaries for revision $revision"
+
+  if [ -d build/$revision ]; then
+    return
+  fi
+
+  temp=$(mktemp -d)
+  git clone . $temp
+  cd $temp
+  git checkout $revision
+  ./tools/devtool -y build --release
+  cd -
+  mkdir -p build/$revision
+  cp -v $temp/build/cargo_target/$(uname -m)-unknown-linux-musl/release/{firecracker,jailer} build/$revision/
+  rm -rf $temp
+}
+
+a_commit=$1
+b_commit=$2
+
+shift 2
+
+say "Performing an A..B-test across $a_commit..$b_commit. This includes the following commits:"
+git --no-pager log --oneline $a_commit..$b_commit
+
+get_binaries $a_commit
+get_binaries $b_commit
+
+./tools/devtool sh "cd tests && PERFORMANCE_TEST_REVISION_A=$a_commit PERFORMANCE_TEST_REVISION_B=$b_commit pytest $@ -m nonci"
```

Comment:
May I ask why you want to always have `-m nonci`? Can't we apply A/B test framework to tests not marked with `nonci`?

Reply:
Mh, good catch, no need to pass this here (if someone wants to run a `nonci` test, they can just pass `-m nonci` via script arguments.